src/intelligence_layer/connectors/retrievers/document_index_retriever.py

```diff
@@ -20,16 +20,17 @@
 class DocumentIndexRetriever(BaseRetriever[DocumentPath]):
     """Search through documents within collections in the `DocumentIndexClient`.
 
-    We initialize this Retriever with a collection & namespace names, and we can find the documents in the collection
-    most semanticly similar to our query.
+    This retriever lets you search for relevant documents in the given Document Index collection.
 
     Args:
-        document_index: Client offering functionality for search.
-        index_name: The name of the index to be used.
-        namespace: The namespace within the `DocumentIndexClient` where all collections are stored.
-        collection: The collection within the namespace that holds the desired documents.
-        k: The (top) number of documents to be returned by search.
-        threshold: The mimumum value of cosine similarity between the query vector and the document vector.
+        document_index: The Document Index client.
+        index_name: The name of the Document Index index to use.
+        namespace: The Document Index namespace.
+        collection: The Document Index collection to use. This is the search context for the retriever.
+        k: The number of most-relevant documents to return when searching.
+        threshold: The minimum score for search results. For semantic indexes, this is the cosine
+            similarity between the query and the document chunk. For hybrid indexes, this corresponds
+            to fusion rank.
```

Comment:
I know this was messed up before, but having the args at class level is not something we should be keeping. The correct way would be to have public attributes where the args are, and move the args simply into the __init__ docstring. An example can be seen here https://github.com/Aleph-Alpha/intelligence-layer-sdk/blob/d75584ba3bd0ebf036d17be25413b08b5793a29c/src/intelligence_layer/connectors/data/data.py#L29

Reply:
k and threshold also have a default now, should should be represented in the docstirng as `...   . Defaults to XYZ.